packages/mdc-list/foundation.js

```diff
@@ -110,83 +120,27 @@ class MDCListFoundation extends MDCFoundation {
     this.useActivatedClass_ = useActivated;
   }
 
-  /** @param {number} index */
+  /** @return {!Index} */
+  getSelectedIndex() {
+    return this.selectedIndex_;
```

Comment:
This is currently out-of-sync on init if a list already has a selected or checked item(s), because we initialize this property to -1 statically.

Reply:
Yes, that is true. This happens with current implementation as well. We should move pre-selection logic to foundation. I'll create a new issue for this.

Reply:
There's no issue in the current implementation because there's no getter, right?

Reply:
Fixed list component to pre-populated selected state for singleSelection, radio and checkbox during initialization.